gateway/tests/tests/fixtures/fed.rs

```diff
@@ -58,11 +64,20 @@ impl IFederationApi for MockApi {
     }
 
     async fn fetch_gateways(&self) -> Result<Vec<LightningGateway>, ApiError> {
-        unimplemented!()
+        if let Some(gateway) = &*self
+            .gateway
+            .lock()
+            .expect("failed to read registered gateways")
+        {
+            Ok(vec![gateway.clone()])
+        } else {
+            Ok(vec![])
+        }
```

Comment:
This could probably be rewritten to

```rust
self
    .gateway
    .lock()
    .expect("failed to read registered gateways")
    .into_iter()
    .cloned()
    .collect::<Vec<_>>()
```